projects/samples/contests/robocup/protos/RobocupSoccerField.proto

```diff
@@ -11,282 +12,267 @@ PROTO RobocupSoccerField [
+    cin_start = Npts - 2 * n_vertices_quarter;
+    cout_start = Npts - n_vertices_quarter;
 
-    -- External triangles
-    Ntg = table.maxn(tg) -- number of triangles for grass already existing
-    tg[Ntg + 1] = {a = 33, b = cout_start + 1, c = 40} -- adds first triangle (boundary with half line)
-    Ntg = table.maxn(tg)
-    for i = 1, (n_vertices_quarter - 1) do
-      tg[Ntg + i] = {a = 33, b = cout_start + (i + 1), c = cout_start + i}
-    end
-    tg[Ntg + n_vertices_quarter] = {a = 33, b = 35, c = cout_start + n_vertices_quarter} -- adds last triangle (boundary with penalty area and touch line)
+    // External triangles
+    tg.push({a: 33, b: cout_start + 1, c: 40}) // adds first triangle (boundary with half line)
+    for (let i = 1; i <= n_vertices_quarter - 1; ++i)
+      tg.push({a: 33, b: cout_start + (i + 1), c: cout_start + i});
 
-    -- Internal triangles
-    -- There is a vertex junction_index of the inner circle (cin) that makes the junction between triangles
-    -- going through pt.41 (top vertex of central dash) and pt.36 (bottom vertex of central dash).
-    -- It is located at approximately two thirds of the quarter
-    junction_index = math.ceil((2 / 3) * n_vertices_quarter) -- index of the junction triangle within the inner circle
-    Ntg = table.maxn(tg)
-    for i = 1, (junction_index - 1) do
-      tg[Ntg + i] = {a = cin_start + (i + 1), b = 41, c = cin_start + i}
-    end
-    tg[Ntg + junction_index] = {a = cin_start + junction_index, b = 36, c = 41} -- adds junction triangle
-    for i = junction_index, (n_vertices_quarter - 1) do
-      tg[Ntg + i + 1] = {a = cin_start + (i + 1), b = 36, c = cin_start + i}
-    end
-    tg[Ntg + n_vertices_quarter + 1] = {a = cin_start + n_vertices_quarter, b = 37, c = 36} -- adds touch line triangle (boundary with touch line)
+    tg.push({a: 33, b: 35, c: cout_start + n_vertices_quarter}); // Adds last triangle (boundary with penalty area and touch line)
 
-    -- Replication of the other 3 quarters
-    Npts = table.maxn(pts)
-    for i = 1, Npts do
-      pt_ref = pts[i] -- reference point
-      -- 2nd Quarter (which completes one team's half)
-      pts[Npts + i] = {x = pt_ref.x, y = 2 * I + B - pt_ref.y, z = pt_ref.z}
-      -- 3rd Quarter (replication on the y-axis)
-      pts[2 * Npts + i] = {x = 2 * I + A - pt_ref.x, y = pt_ref.y, z = pt_ref.z}
-      -- 4th Quarter (which completes opponent's half)
-      pts[3 * Npts + i] = {x = 2 * I + A - pt_ref.x, y = 2 * I + B - pt_ref.y, z = pt_ref.z}
-    end
+    // Internal triangles
+    // There is a vertex junction_index of the inner circle (cin) that makes the junction between triangles
+    // going through pt.40 (top vertex of central dash) and pt.35 (bottom vertex of central dash).
+    // It is located at approximately two thirds of the quarter
+    let junction_index = Math.ceil((2 / 3) * n_vertices_quarter); // Index of the junction triangle within the inner circle
```

Comment:
This can be const
```suggestion
    const junction_index = Math.ceil((2 / 3) * n_vertices_quarter); // Index of the junction triangle within the inner circle
```